core/runtime/src/main/java/io/quarkus/runtime/logging/BannerFormatter.java

```diff
@@ -58,6 +59,13 @@ public String format(ExtLogRecord extLogRecord) {
 
     @Override
     public String getHead(Handler h) {
-        return bannerSupplier.get();
+        final String banner = bannerSupplier.get();
+        final Charset charset;
+        try {
+            charset = Charset.forName(h.getEncoding());
+            return charset.newEncoder().canEncode(banner) ? banner : "";
```

Comment:
should we put a "(banner can not be encoded on " + charset + ")" or at least a debug log ?

Reply:
The challenge is that you can't really log from here. I think we could enhance the log manager to do that though at some point.